fetcher.go

```diff
@@ -48,6 +51,16 @@ var (
 func (p *pool) doFetch(ctx context.Context, from string, c cid.Cid, attempt int) (b blocks.Block, rm tieredhashing.ResponseMetrics, e error) {
 	reqUrl := fmt.Sprintf(saturnReqTmpl, c)
 
+	rand, _ := rand.Int(rand.Reader, big.NewInt(sentinelCidPeriod))
```

Comment:
This should be in fetch Resource as that is called for both blocks and CARs.

Reply:
rather than spawning a go-routine / adding a separate percentage / code path for these sentinals, can we merge this with the mirroring of requests and subsequent validation that happens in `checkPool`?